server/controllers/finance/priceList.js

```diff
@@ -229,6 +236,92 @@ exports.createItem = function createItem(req, res, next) {
     .done();
 };
 
+
+/**
+ * @method downloadTemplate
+ *
+ * @description send to the client the template file for price list item import
+*/
+exports.downloadTemplate = (req, res, next) => {
+  try {
+    const file = path.join(__dirname, '../../resources/templates/import-inventory-item-template.csv');
+    res.download(file);
+  } catch (error) {
+    next(error);
+  }
+};
+
+exports.downloadFilledTemplate = async (req, res, next) => {
+  try {
+    const pdfOptions = {
+      orientation : 'landscape',
+      footerRight : '[page] / [toPage]',
+      footerFontSize : '7',
+      csvKey : 'rows',
+      suppressDefaultFormatting : true,
+      suppressDefaultFiltering : true,
+    };
+    const optionReport = _.extend(req.query, pdfOptions, {
+      filename : 'TREE.STOCK_VALUE',
+    });
+    const report = new ReportManager('', req.session, optionReport);
+    const sql = 'SELECT code, text FROM inventory';
+    const rows = await db.exec(sql);
+    const data = rows.map(row => {
+      row.price = '';
+      row.is_percentage = '';
+      return row;
+    });
+    const result = await report.render({ rows : data }, null, { csvKey : 'rows' });
+    res.set(csv.headers).send(result.report);
+  } catch (ex) {
+    next(ex);
+  }
+
+};
+
+exports.importItem = async (req, res, next) => {
+  try {
+    if (!req.files || req.files.length === 0) {
+      const errorDescription = 'Expected at least one file upload but did not receive any files.';
+      throw new BadRequest(errorDescription, 'ERRORS.MISSING_UPLOAD_FILES');
+    }
+
+    const filePath = req.files[0].path;
+
+    const data = await util.formatCsvToJson(filePath);
+    removeEmptyLines(data);
+    if (!hasValidDataFormat(data)) {
+      throw new BadRequest('The given file has a bad data format for stock', 'ERRORS.BAD_DATA_FORMAT');
+    }
+    const priceListUuid = db.bid(req.body.pricelist_uuid);
+    const sql = 'CALL importPriceListItem(?,?,?,?);';
+    const transaction = db.transaction();
+    data.forEach(item => {
+
+      transaction.addQuery(sql, [priceListUuid, item.code, item.price, item.is_percentage]);
+    });
+    await transaction.execute();
+    res.sendStatus(200);
+  } catch (ex) {
+    next(ex);
+  }
+
+};
+
+function removeEmptyLines(data) {
+  data.forEach((r, index) => {
+    if (!(r.code && r.price)) delete data[index];
```

Comment:
Did you know that when you delete an array element, the length isn't affected?  Try it:
```js
const array = [1, 2, 3, 4, 5];
console.log(array.length); // => 5
delete array[2];
console.log(array.length); // => 5
```

The `delete` operator also has some pretty strange behavior for prototype versus non prototype objects, strict vs non-strict mode, etc.  You can read all about it in [the docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete).

For arrays, I always prefer to simply `filter()`.  Like so:
```js
return data.filter(row => (row.code && r.price));
```